src/hooks/core/chat/useFetchChannelHook.ts

```diff
@@ -103,7 +111,11 @@ const useFetchChannelHook = () => {
 
       await Promise.all(
         (channel?.messages || []).map(async (message) => {
-          if (message?.type === 'deleted') return;
+          const isDeletedMessage = message?.message_type === MESSAGE_TYPE_DELETED;
+          const isDeletedHelper = Boolean(message?.deleted_message_id);
+          if (isDeletedMessage && isDeletedHelper) return;
+
+          if (isDeletedMessage) message.text = DELETED_MESSAGE_TEXT;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to the previous comment, modifying the `message.text` directly could lead to potential issues. Consider using a new property for the display text.
```diff
-    const isDeletedMessage = message?.message_type === MESSAGE_TYPE_DELETED;
-    if (isDeletedMessage) message.text = DELETED_MESSAGE_TEXT;
+    const isDeletedMessage = message?.message_type === MESSAGE_TYPE_DELETED;
+    if (isDeletedMessage) message.displayText = DELETED_MESSAGE_TEXT;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->